### PR TITLE
[Integration Tests] Mempool additional tests 2

### PR DESCRIPTION
**Pull request by zeptin**
Several of these are placeholders until the relevant issues are fixed.

### CODE REVIEW

File: src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs [hunk 1]
```diff
@@ -13,32 +18,114 @@ namespace Stratis.Bitcoin.Features.MemoryPool.Tests
...
-    /// <remarks>TODO: Currently only stubs - need to complete</remarks>
     public class MempoolValidatorTest : TestBase
     {
         public MempoolValidatorTest() : base(KnownNetworks.RegTest)
         {
         }
 
-        [Fact(Skip = "Not implemented yet.")]
-        public void CheckFinalTransaction_WithStandardLockTimeAndValidTxTime_ReturnsTrue()
+        [Fact(Skip = "Awaiting fix for issue #2474")]
```
Comment (dangershony) [inline src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: Are you still waiting?

Comment (zeptin) [reply src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: Going to work on the fix myself I think

Comment (dangershony) [reply src/Stratis.Bitcoin.Features.MemoryPool.Tests/MempoolValidatorTest.cs]: I just did that dont see PRs

**Comment by dangershony** [comment]
See here https://github.com/stratisproject/StratisBitcoinFullNode/issues/2474

**Comment by dangershony** [comment]
One thing that is missing here is to check that we do not including invalid trx from mempool when we mine a block, we need to bring this code from stratisX
```
         // Timestamp limit
            if (tx.nTime > GetAdjustedTime() || (fProofOfStake && tx.nTime > pblock->vtx[0].nTime))
                continue;
```

**Comment by dangershony** [comment]
Fix: #2027

**Pull request merged**